### PR TITLE
[bitvec] Move to common for usage outside network module

### PR DESCRIPTION
**Pull request by bothra90**
The library has also been modified to use Vec<u8> instead of Bytes to avoid dependency on the Bytes crate.

Bytes provides COW (copy-on-write) semantics and faster serialization/deserialization by consuming the underlying byte buffer in one shot instead of one-by-one as a sequence. However, since bitvec is not expected to be used in perf-critical code paths, it's ok for now to
use Vec.

Another alternative for fast serde is `serde_bytes` which can be explored in the future if the need for more performant serialization/deserialization arises.

### CODE REVIEW

File: common/bitvec/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "bitvec"
```
Comment (kchalkias) [inline common/bitvec/Cargo.toml]: Do we want to rename this to bitvector or something else to make it easy to distinguish between this and the other popular bitvec crate?

Comment (bothra90) [reply common/bitvec/Cargo.toml]: @kchalkias: libra-bitvec?

**Comment by metajack** [comment]
`bytes` is currently used in the TCB. Are we doing this because we think we'll be able to remove it?

`libra-crypto` for example uses both `bitvec` and `bit-vec` directly.

I'm not seeing a great justification for why we wrote our own here, or why we're trying to avoid `bytes`.

**Comment by bothra90** [comment]
@metajack : the crypto team is trying to move to this bitvec as well. See #2993. Regarding whether we want to fully remove the dependency on `Bytes`, I'll defer to @davidiw

**Pull request closed without merging**